Atomic/syscontainers.py

```diff
@@ -920,23 +934,17 @@ def _do_checkout(self, repo, name, img, upgrade, deployment, values, destination
         if system_package == 'auto':
             system_package = "yes" if self._should_be_installed_rpm(exports) else 'no'
 
-        manifest_file = os.path.join(exports, "manifest.json")
         installed_files_template = []
         has_container_service = True
         rename_files = {}
-        manifest = None
-        if os.path.exists(manifest_file):
-            with open(manifest_file, "r") as f:
-                try:
-                    manifest = json.loads(f.read())
-                except ValueError:
-                    raise ValueError("Invalid manifest.json file in image: {}.".format(img))
-                if "installedFilesTemplate" in manifest:
-                    installed_files_template = manifest["installedFilesTemplate"]
-                if "renameFiles" in manifest:
-                    rename_files = manifest["renameFiles"]
-                if "noContainerService" in manifest and manifest["noContainerService"]:
-                    has_container_service = False
+        manifest = SystemContainers._read_manifest_file(exports, img)
+
+        if "installedFilesTemplate" in manifest:
```

Comment:
For three local variables here, I am planning to remove them in the future. In order to achieve that, I am planning to do the following steps:
- For all the code blocks that do use it, I am planning to refactor those as small functions. 

e.g: I first wanted to have
```
def _get_manifest_attributes(manifest, str):
      if str in manifest:
          return manifest[str]
```
and for the following:
```
 if has_container_service and upgrade:
            if was_service_active:
                self._systemctl_command("stop", name)
            if os.path.exists(tmpfilesout):
                try:
                    self._systemd_tmpfiles("--remove", tmpfilesout)
                except subprocess.CalledProcessError:
pass
```
can be refactored into a function with 
```
has_container_service = !(_get_manifest_attributes(manifest, "noContainerService"))
```

By removing those three local variables, I believe we can get 2 or more benefits. 
- 1: we reduce the number of local parameters that will be passed into functions
- 2: we shrink https://github.com/projectatomic/atomic/blob/master/Atomic/syscontainers.py#L992-L1006, down the parameter list for writing to info file
- IMO, it might make the code looks nicer

**But, this is only my personal preference, so I would like to hear comments about this, and thanks in advance for the feedback :p**

Reply:
```
 argument of type 'NoneType' is not iterable
Traceback (most recent call last):
  File "/usr/bin/atomic", line 185, in <module>
    sys.exit(_func())
  File "/usr/lib/python3.6/site-packages/Atomic/install.py", line 134, in install
    return be.install(self.image, self.name)
  File "/usr/lib/python3.6/site-packages/Atomic/backends/_ostree.py", line 128, in install
    return self.syscontainers.install(image, name)
  File "/usr/lib/python3.6/site-packages/Atomic/syscontainers.py", line 410, in install
    return_value = self._install(image, name)
  File "/usr/lib/python3.6/site-packages/Atomic/syscontainers.py", line 546, in _install
    self._checkout(repo, name, image, 0, False, values=values, remote=self.args.remote, system_package=self.args.system_package)
  File "/usr/lib/python3.6/site-packages/Atomic/syscontainers.py", line 650, in _checkout
    installed_files_checksum=installed_files_checksum, system_package=system_package)
  File "/usr/lib/python3.6/site-packages/Atomic/syscontainers.py", line 942, in _do_checkout
    if "installedFilesTemplate" in manifest:
TypeError: argument of type 'NoneType' is not iterable
```
hmm, I think this could be fixed by checking manifest to be not None, but since I want to hear comments about the proposal above, and pushing a fixup will hide the comment, I will push a fixup later. (hopefully you won't mind :p)